### DISCUSSION TITLE
When to use the access token?

### DISCUSSION THREAD

**Discussion by niraj-bits** [dotnet/SqlClient#1665] [Q&A] [answered]
Hello, the Sqlclient supports passing in access token, this is retrieved from a REST api call to an endpoint accessible within the Azure VM. I can also specify Authentication=Active Directory MSI in the connection string.

I am confused when should I use the access token? Why is it provided in first place when one can always specify  Authentication=Active Directory MSI, or is it that under the hood, the Sql client gets the access token by itself and sets it.

Thanks
Niraj

**Comment by ErikEJ** [comment, marked as answer]
Yes, the SqlClient calls the Endpoint to get the token "automagically" when you use Azure AD auth options in the connection string.

**Reply by niraj-bits** [reply]
Thank you Erik for confirmation. That's what I thought.
